.github/workflows/build-riscv-docker.yml

```diff
@@ -0,0 +1,69 @@
+name: Build and Package
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ "main", "dev" ]
+
+jobs:
+  build_docker_image:
+    name: Build Docker Image for CI
+    if: github.repository_owner == 'CTSRD-CHERI'
+    runs-on: ubuntu-24.04
+    permissions:
+      contents: read
+      packages: write
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v6
+
+    - name: Install Dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y ninja-build cmake build-essential lld
+    - name: ccache
+      uses: hendrikmuhs/ccache-action@v1.2
+
+    - name: Configure CMake
+      run: |
+        # Build a minimal version of LLVM with features we do not need disabled
+        cmake -G Ninja -S llvm -B build \
+          -DLLVM_ENABLE_ASSERTIONS=TRUE \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DCMAKE_INSTALL_PREFIX=$GITHUB_WORKSPACE/install \
+          -DLLVM_ENABLE_PROJECTS="clang;lld" \
+          -DLLVM_TARGETS_TO_BUILD="RISCV;X86" \
```

Comment:
Why X86 but not say AArch64? Just because it's the native architecture for the container being built?